whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFullTextRebuild.java

```diff
@@ -0,0 +1,263 @@
+import java.util.stream.Collectors;
+
+import static net.ripe.db.whois.api.elasticsearch.ElasticSearchConfigurations.getMappings;
+import static net.ripe.db.whois.api.elasticsearch.ElasticSearchConfigurations.getSettings;
+
+@Component
+public class ElasticFullTextRebuild {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ElasticFullTextRebuild.class);
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss");
+
+    private final ElasticIndexService elasticIndexService;
+
+    private static final int BATCH_SIZE = 100;
+    private final JdbcTemplate jdbcTemplate;
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private final String source;
+    private final List<String> elasticHosts;
+
+    @Autowired
+    public ElasticFullTextRebuild(final ElasticIndexService elasticIndexService,
+                                  @Value("#{'${elastic.host:}'.split(',')}") final List<String> elasticHosts,
+                                  @Qualifier("whoisSlaveDataSource") final DataSource dataSource,
+                                  @Value("${whois.source}") final String source) {
+        this.elasticIndexService = elasticIndexService;
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.source = source;
+        this.elasticHosts = elasticHosts;
+        this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(jdbcTemplate);
+    }
+
+    public void run() throws IOException {
+        LOGGER.info("Source is  {}", source);
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+
+        final String indexName = "whois-" + DATE_TIME_FORMATTER.format(LocalDateTime.now());
+
+        rebuild(indexName, elasticIndexService.getMetadataIndex(), true);
+
+        deleteOldIndexes(indexName);
+
+        LOGGER.info("ES indexing complete {}", stopwatch);
+    }
+
+    public void rebuild(final String indexName, final String metadataName, final boolean shouldSetAlias) throws IOException {
+
+        final RestHighLevelClient client = elasticIndexService.getClient();
+
+        LOGGER.info("Start building index {}", indexName);
+
+        createIndex(indexName);
+
+        final int maxSerial = JdbcRpslObjectOperations.getSerials(jdbcTemplate).getEnd();
+        LOGGER.info("Indexing upto serial number {}", maxSerial);
+
+        final List<List<Integer>> objectIds = Lists.partition(jdbcTemplate.queryForList("select object_id from last where sequence_id > 0", Integer.class), BATCH_SIZE);
+
+        final Timer timer = new Timer(true);
+        final AtomicInteger totalProcessed = new AtomicInteger(0);
+        printProgress(totalProcessed, objectIds.stream().mapToInt(i -> i.size()).sum(), timer);
+
+        final List<Integer> failedToIndexed = Lists.newArrayList();
+        final List<CIString> failedToParse = Lists.newArrayList();
+
+        objectIds.parallelStream().forEach(batchObjectIds -> {
```

Comment:
Be careful with parallelStream may consume all available threads, can you limit to a fixed pool ?

Reply:
Good idea